src/strings/utf8.c

```diff
@@ -353,7 +353,7 @@ MVMuint32 MVM_string_utf8_decodestream(MVMThreadContext *tc, MVMDecodeStream *ds
     MVMDecodeStreamBytes *cur_bytes = NULL;
     MVMDecodeStreamBytes *last_accept_bytes     = ds->bytes_head;
     MVMDecodeStreamBytes *lag_last_accept_bytes = NULL;
-    MVMint32 last_accept_pos, lag_last_accept_pos, ready, at_start;
+    MVMint32 last_accept_pos, lag_last_accept_pos = 0, ready, at_start;
```

Comment:
In what case can `lag_last_accept_pos` actually be used uninitialized like the compiler warns us about and why is initializing with 0 the right thing to do as compared to some change to the algorithm?

Reply:
There are combinations of conditionals that could lead to it being used uninitialized. I haven't checked whether those combinations are actually possible. @jnthn `lag_last_accept_pos` was introduced in https://github.com/MoarVM/MoarVM/commit/a6abd3c6654413d2230470dbaa82b7b3a2b05762, do you have a suggestion?

Reply:
I think it hinges on the use of `-1` in `lag_codepoint` being used as an indication of whether there'll be something valid in `lag_last_accept_pos`. It may be that the compiler can't track that.

Reply:
Ah, so you think initializing to 0 to silence the warning is acceptable?